src/Orleans.Core/Messaging/DnsNameGatewayListProvider.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Orleans.Configuration;
+using Orleans.Runtime;
+
+namespace Orleans.Messaging
+{
+    /// <summary>
+    /// This Gateway list provider looks up ip addresses based on a dns
+    /// name.  This is ideal for container environments (kubernetes, swarm)
+    /// as well as Service Fabric using the dns feature.
+    /// </summary>
+    public class DnsNameGatewayListProvider : IGatewayListProvider
+    {
+        private readonly DnsNameGatewayListProviderOptions options;
+        private readonly TimeSpan maxStaleness;
+        private readonly ILoggerFactory loggerFactory;
```

Comment:
Not very important: You don't use the loggerFactory in the class at all, you could remove this field